assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -149,6 +178,51 @@ const baseActions = {
 
 		return { response, error };
 	},
+
+	/**
+	 * Registers a module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}    slug                         Module slug.
+	 * @param {Object}    [settings]                   Optional. Module settings.
+	 * @param {string}    [settings.name]              Optional. Module name. Default is the slug.
+	 * @param {string}    [settings.description]       Optional. Module description. Default empty string.
+	 * @param {string}    [settings.icon]              Optional. Module icon. Default empty string.
+	 * @param {number}    [settings.order]             Optional. Numeric indicator for module order. Default 10.
+	 * @param {string}    [settings.homepage]          Optional. Module homepage URL. Default empty string.
+	 * @param {WPElement} [settings.settingsComponent] React component to render the settings panel. Default is the DefaultModuleSettings component.
+	 * @return {Object} Redux-style action.
+	 */
+	*registerModule( slug, { settingsComponent = DefaultModuleSettings, ...settings } = {} ) {
+		invariant( slug, 'module slug is required' );
+
+		const registry = yield commonActions.getRegistry();
+		const registryKey = yield registry.select( CORE_SITE ).getRegistryKey();
```

Comment:
@tofumatt Double-checking, is this okay to do? Can we `yield` a selector like this? I would think that in other cases where we need selector data to be available we've been using `waitFor*` controls - is that different here?

Reply:
This doesn't immediately strike me as something that should work, because `getRegistryKey()` is not a generator function and thus will immediately return. It will _seem_ like it works because `getRegistryKey()` returns a value, but if it were a selector with a resolver, this wouldn't wait until the resolver has completed—at least to my knowledge.

Good catch; I think this needs to be refactored to wait until `registryKey !== undefined`. (https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/analytics/datastore/tags.js#L102 is an example of that sort of wait control, for reference.)

If this _does_ work and there's a test that covered it that'd be cool, but from my knowledge of `@wordpress/data` internals this shouldn't work, but either passes tests because it's very fast or we aren't actually testing it. It's possible that it's continuing on with `undefined` as the key and that's working.